### PR TITLE
feat: Add cloud service connection support

### PR DESCRIPTION
**Pull request by pento**
## Summary

- **REST endpoints**: `POST/GET/DELETE /wpce/v1/cloud` — the cloud service pushes its URL and API key to the plugin during the setup wizard
- **Editor JS**: on editor load, if cloud settings exist, calls `POST {cloudUrl}/api/v1/connect` with the API key — this triggers the cloud service to create a SessionManager and connect via Yjs sync
- **Onboarding UI**: shows "Connecting to Claudaborative Cloud..." spinner when cloud is configured, hides setup modal

Companion PR: pento/claudaborative-cloud#5

## Test plan

- [ ] Cloud setup wizard auto-pushes settings to plugin
- [ ] Opening editor triggers cloud connect call
- [ ] MCP connection detected via Yjs after cloud connects
- [ ] Onboarding content shows spinner instead of setup instructions
- [ ] Setup modal does not appear when cloud is configured
- [ ] DELETE /wpce/v1/cloud removes settings

### CODE REVIEW

File: wordpress-plugin/src/cloud/connect.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+	const url = `${state.cloudUrl.replace(/\/+$/, '')}/api/v1/connect`;
+
+	fetch(url, {
+		method: 'POST',
+		headers: {
+			Authorization: `Bearer ${state.cloudApiKey}`,
+		},
+		// No credentials -- cross-origin request with API key auth.
+		mode: 'cors',
```
[the remark this answers is not in the document]
Comment (pento) [reply wordpress-plugin/src/cloud/connect.ts:67]: Fixed in cb8f0be. Added a client-side HTTPS guard in `connectToCloud()` that refuses to send the Bearer token unless the URL scheme is `https:` (with an exception for `localhost`, `127.0.0.1`, and `[::1]` for dev). Invalid URLs are also caught and silently bail. This is defense-in-depth alongside the server-side validation on the REST endpoint.

File: wordpress-plugin/claudaborative-editing.php [hunk 2]
```diff
@@ -100,9 +100,18 @@ public static function enqueue_editor_assets() {
...
 		wp_add_inline_script(
 			'wp-hooks',
-			'window.wpceInitialState = ' . wp_json_encode( array( 'mcpConnected' => $mcp_connected ) ) . ';' .
+			'window.wpceInitialState = ' . wp_json_encode(
+				array(
+					'mcpConnected' => $mcp_connected,
+					'cloudUrl'     => $cloud_url,
+					'cloudApiKey'  => $cloud_api_key,
+				)
+			) . ';' .
```
[the remark this answers is not in the document]
Comment (pento) [reply wordpress-plugin/claudaborative-editing.php:114]: Acknowledged — this is a known tradeoff in the current design. The API key here is a site-level token (not a per-user credential) that the cloud service pushes during site setup. It's used for a single idempotent `POST /api/v1/connect` call that tells the cloud service to initiate Yjs sync. The alternative (server-side connect on page load) would add blocking HTTP requests to the critical path of every editor load and introduce timeout/retry complexity.

That said, moving to short-lived per-user tokens is a good future improvement. For now, the key is scoped: it only authorizes a connect call to the specific cloud service, and the new HTTPS validation (cb8f0be) ensures it can't leak over plaintext.

File: wordpress-plugin/includes/class-rest-controller.php [hunk 3]
```diff
@@ -226,6 +261,15 @@ public function edit_posts_permissions( $request ) {
 		return true;
 	}
 
+	/**
+	 * Permission check: user can manage options (administrator).
+	 *
+	 * @return bool True if permitted.
+	 */
+	public function manage_options_permissions() {
```
[the remark this answers is not in the document]
Comment (pento) [reply wordpress-plugin/includes/class-rest-controller.php]: Fixed in cb8f0be. Added the `$request` parameter to `manage_options_permissions()` for consistency with the other permission callbacks in the class. PHPStan and PHPCS both pass.

File: wordpress-plugin/includes/class-rest-controller.php [hunk 4]
```diff
@@ -163,6 +163,41 @@ public function register_routes() {
...
+					'callback'            => array( $this, 'update_cloud_settings' ),
+					'permission_callback' => array( $this, 'manage_options_permissions' ),
+					'args'                => array(
+						'cloud_url' => array(
+							'required'          => true,
+							'type'              => 'string',
+							'format'            => 'uri',
+							'sanitize_callback' => 'sanitize_url',
+						),
+						'api_key'   => array(
```
[the remark this answers is not in the document]
Comment (pento) [reply wordpress-plugin/includes/class-rest-controller.php:187]: Fixed in cb8f0be. Added a `validate_cloud_url()` callback on the `cloud_url` parameter that requires HTTPS, with an exception for `localhost`/`127.0.0.1`/`::1` for local development. Non-HTTP schemes (ftp, etc.) are also rejected. PHP tests cover all branches: HTTPS allowed, HTTP rejected, localhost HTTP allowed, 127.0.0.1 HTTP allowed, and FTP rejected.

File: wordpress-plugin/includes/class-rest-controller.php [hunk 5]
```diff
@@ -738,6 +824,54 @@ public function get_sync_entity_single( $request ) {
...
+	 * via wpceInitialState (only visible to authenticated users).
+	 *
+	 * @return \WP_REST_Response Response.
+	 */
+	public function get_cloud_settings() {
+		$cloud_url = get_option( 'wpce_cloud_url', '' );
+
+		return rest_ensure_response(
+			array(
+				'configured' => ! empty( $cloud_url ),
```
[the remark this answers is not in the document]
Comment (pento) [reply wordpress-plugin/includes/class-rest-controller.php]: Fixed in 183bf85. `get_cloud_settings()` now checks both `wpce_cloud_url` and `wpce_cloud_api_key` are non-empty before reporting `configured: true`, matching the client-side `isCloudConfigured()` logic. Added a test for the partial-config edge case (URL set, key missing).
